File: web_automator.py
import sys
import webbrowser

URLS = {
    "work":["https://www.slack.com","https://www.google.com","https://www.10web.com","https://docs.python.org/3/library/webbrowser.html"],
    "personal":["https://www.youtube.com","https://hianime.to","https://www.github.com","https://www.spotify.com"]
}

def open_webpages(urls):
    for url in urls:
        webbrowser.open(url)
# 
if __name__ == "__main__":
    if len(sys.argv) != 2 or sys.argv[1] not in URLS:
        print("Usage: python script.py <set_name>")
        print("Available sets:")
        for set_name in URLS.keys():
            print(f"- {set_name}")
        sys.exit(1)

    set_name = sys.argv[1]
    urls = URLS[set_name]
    open_webpages(urls)

File: web_automator.js
# npm install open
const open = require('open');
const process = require('process');

const URLS = {
    work: [
        "https://www.slack.com",
        "https://www.google.com",
        "https://www.10web.com",
        "https://docs.python.org/3/library/webbrowser.html"
    ],
    personal: [
        "https://www.youtube.com",
        "https://hianime.to",
        "https://www.github.com",
        "https://www.spotify.com"
    ]
};

function openWebpages(urls) {
    urls.forEach(url => open(url));
}

if (process.argv.length !== 3 || !URLS[process.argv[2]]) {
    console.log("Usage: node script.js <set_name>");
    console.log("Available sets:");
    for (const setName in URLS) {
        console.log(`- ${setName}`);
    }
    process.exit(1);
}

const setName = process.argv[2];
const urls = URLS[setName];
openWebpages(urls);
